extension/lib/bigcommerce/CartRepository.js

```diff
@@ -93,6 +103,39 @@ class BigCommerceCartRepository {
     return lineItem
   }
 
+  /**
+   * @param {[BigCommerceCartLineItemUpdateRequest]} items
+   * @param {bigCommerceUpdateFailureNotifier} updateFailureNotifier
+   * @return {Promise<void>}
+   */
+  async updateItems (items, updateFailureNotifier) {
+    const cart = await this.load()
```

Comment:
We have a chance of getting a null here.

Reply:
Added check for null cart. If cart is null the message the updateFailureNotifier function is called

Reply:
My local repository was ahead of the remote repository. I did a push. I checked the local repository and saw the correction the JSDoc.

Reply:
In this case it is a hard error for sure, but let's wait a bit until we clear up the way we will process soft errors.

Reply:
I feel like after our conversation today we have the soft fail handling figured out. I added a hard fail for situations when no cart is delivered.